snews_cs/cs_utils.py

```diff
@@ -107,84 +107,121 @@ def is_garbage_message(snews_message, is_test=False):
+        log.debug("Hard reset or connection test.  Skipping format check.\n")
+        return True
+
+    elif '_Heartbeat_' in snews_message['_id']:
+        if snews_message["detector_status"] in ["ON","OFF"]:
+            log.debug("Heartbeat message.  Skipping format check.\n")
+            return True
+        else:
+            warning += f"Expected detector status to be 'ON' or 'OFF' got {snews_message['detector_status']}"
+            print(warning)
+            log.warning(warning)
+            return False
+
     detector_file = os.path.abspath(os.path.join(os.path.dirname(__file__), 'auxiliary/detector_properties.json'))
     with open(detector_file) as file:
         snews_detectors = json.load(file)
     snews_detectors = list(snews_detectors.keys())
-    message_key = snews_message.keys()
-    is_garbage = False
-    missing_key = False
-    warning = f'The following Message does not meet SNEWS CS standards !\n{snews_message}\n'
-    if 'detector_name' not in message_key:
+
+    if 'detector_name' not in message_keys:
         warning += f'* Does not have required key: "detector_name"\n'
-        is_garbage = True
+        snews_format = False
         missing_key = True
-    if 'neutrino_time' not in message_key:
+
+    elif snews_message['detector_name'] not in snews_detectors:
+        warning += f'* Detector not found: {snews_message["detector_name"]}\n'
+        warning += f"Detector options: {snews_detectors}\n"
+        snews_format = False
+
+    # Check for missing keys
+    if 'neutrino_time' not in message_keys:
         warning += f'* Does not have required key: "neutrino_time"\n'
-        is_garbage = True
         missing_key = True
-    # if 'p_val' not in message_key:
-    #     warning += f'* Does not have required key: "p_val"\n'
-    #     is_garbage = True
-    #     missing_key = True
+        snews_format = False
+
     if missing_key:
         log.warning(warning)
-        return is_garbage
-    contents_suck = False
-    if type(snews_message['p_val']) is not float:
-        contents_suck = True
-        is_garbage = True
-        warning += f'* p value needs to be a float type, type given: {type(snews_message["p_val"])}\n'
-    if type(snews_message['p_val']) is float and (snews_message['p_val'] >= 1.0 or snews_message['p_val'] <= 0):
-        warning += f'* {snews_message["p_val"]} is not a valid p value !\n'
-        contents_suck = True
-        is_garbage = True
-    if type(snews_message['detector_name']) is not str:
-        contents_suck = True
-        is_garbage = True
-        warning += f'* "detector_name" must be a str, type given: {type(snews_message["detector_name"])}\n'
+
+        return snews_format
+ 
+    # Check contents
+    if 'p_val' in message_keys and snews_message['p_val'] is not None:
+        log.debug(f"\nChecking p_val: {snews_message['p_val']}\n")
+        key_type = type(snews_message['p_val'])
```

Comment:
just to make sure; are they really floats or are they in string format due to json? 
I mean they can be `"0.1"` instead of `0.1`

Reply:
I think they are floats but I haven't double-checked.  That check was part of the original garbage checker, which makes me think it works with .json files or we would have noticed by now.  We should probably look into this more though.